homeassistant/components/aemet/const.py

```diff
@@ -200,44 +205,53 @@
     FORECAST_MODE_HOURLY: ATTR_API_FORECAST_HOURLY,
 }
 
-FORECAST_SENSOR_TYPES: tuple[SensorEntityDescription, ...] = (
-    SensorEntityDescription(
+
+@dataclass
+class ForecastSensorEntityDescription(SensorEntityDescription):
+    """Class describing Aemet forecast sensor entities."""
+
+    value: Callable = date
```

Comment:
We need to apply the this before returning the native value. The benefit is that we have a common case for all descriptions that have this attribute instead of a single case for just forecast time.

The default `date` doesn't seem to be correct for all descriptions. Either set a default that works for all descriptions or set a default `None` and check for not `None` before applying the function in `native_value`.

https://github.com/home-assistant/core/blob/08a3140e6cbffc5677971f98191e24eea7085236/homeassistant/components/airly/sensor.py#L171

Reply:
I'm sorry but I've tried and I don't know how to properly do this without errors/warning of any kind.
I'm reverting the latest commit and if anyone wants to move this case to a lambda it will be welcome.